tests/test_wind_energy.py

```diff
@@ -732,6 +732,83 @@ def test_missing_valuation_params(self):
+
+        Returns:
+            None.
+
+        Raises:
+            AssertionError when the two geometries or field values are not
+            equal up to desired precision (default is 6).
+
+        """
+        a_shape = ogr.Open(a_vector_path)
+        a_layer = a_shape.GetLayer(0)
+        a_feat = a_layer.GetNextFeature()
+
+        b_shape = ogr.Open(b_vector_path)
+        b_layer = b_shape.GetLayer(0)
+        b_feat = b_layer.GetNextFeature()
+
+        while a_feat is not None:
+            # Get coordinates from geometry and store them in a list
+            a_geom = a_feat.GetGeometryRef()
+            a_geom_list = re.findall(r'\d+\.\d+', a_geom.ExportToWkt())
+            a_geom_list = [float(x) for x in a_geom_list]
+
+            b_geom = b_feat.GetGeometryRef()
+            b_geom_list = re.findall(r'\d+\.\d+', b_geom.ExportToWkt())
+            b_geom_list = [float(x) for x in b_geom_list]
+
+            try:
+                # The coefficient of 1.5 here derives from when
+                # `assert_array_almost_equal` was used, which had parameter
+                # `decimal=4`. In the numpy implementation, this meant an
+                # absolute tolerance of 1.5 * 10**-4.
+                # In other places we were able to round 1.5 down to 1,
+                # but here the slightly larger tolerance is needed.
+                numpy.testing.assert_allclose(
+                    a_geom_list, b_geom_list, rtol=0, atol=1.5e-4)
+            except AssertionError:
+                a_feature_fid = a_feat.GetFID()
+                b_feature_fid = b_feat.GetFID()
+                raise AssertionError('Geometries are not equal in feature %s, '
+                                     'regression feature %s.' %
+                                     (a_feature_fid, b_feature_fid))
+
+            # Get field names/values as dictionaries and compare them without
+            # specifying
+            a_fields = a_feat.items()
+            b_fields = b_feat.items()
+            for a_field, a_value in a_fields.items():
+                try:
+                    b_value = b_fields[a_field]
+                except KeyError:
+                    raise AssertionError(
+                        'Field %s in feature %s does not exist in regression'
+                        'feature %s.' % (a_field, a_feature_fid, b_feature_fid))
+                try:
+                    numpy.testing.assert_allclose(a_value, b_value, rtol=0, atol=1e-6)
+                except AssertionError:
+                    raise AssertionError(
+                        'Values in %s field are not equal in feature %s: %s, '
+                        'regression feature %s: %s.' %
+                        (a_field, a_feature_fid, a_value, b_feature_fid, b_value))
+
+            a_feat = None
+            b_feat = None
+            a_feat = a_layer.GetNextFeature()
+            b_feat = b_layer.GetNextFeature()
+
+        a_shape = None
+        b_shape = None
+

```

Comment:
The wholesale inclusion of this function (like with in `test_hra.py` and `test_recreation.py` makes me wonder if `git` included the function by mistake in the merge.  Is this function used in the file?  Is it needed?

Reply:
Oh, thanks for catching these! Yes it must have been a result of the merge.